### PR TITLE
Add class type for underlying message

### PR DESCRIPTION
**Pull request by mfine**
Add a classtype to expose the underlying message.

### CODE REVIEW

File: haskell/src/Data/RTCM3/TH.hs [hunk 1]
```diff
@@ -22,8 +22,8 @@ import Language.Haskell.TH
 deriveRTCM3 :: Name -> Q [Dec]
 deriveRTCM3 name =
   [d|instance ToRTCM3 $(conT name) where
-       toRTCM3 msg = Msg len payload crc where
-         payload = toStrict $ encode msg
-         len     = fromIntegral $ length payload
-         crc     = checkCrc len payload
+       toRTCM3 n = Msg len pay crc where
```
Comment (JoshuaGross) [inline haskell/src/Data/RTCM3/TH.hs]: fwiw I think the older variable names were easier to read

**Comment by mookerji** [comment]
👍

File: haskell/src/Data/RTCM3/TH.hs (hunk 1 above)
Comment (mfine) [inline haskell/src/Data/RTCM3/TH.hs]: older variables had name clashes because of `makeClassy`

**Comment by JoshuaGross** [comment]
looks good!

**Pull request merged**